### ISSUE TITLE
Incompatibility with ios17

### ISSUE THREAD

**Issue by kjxbyz** [pichillilorenzo/flutter_inappwebview#1783]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology      | Version       |
|-----------------| ------------- |
| Flutter version |       3.13.4        |
| Plugin version  |       5.7.2+3        |
| Android version |               |
| iOS version     |        17       |
| macOS version   |       13.5.2         |
| Xcode version   |         15      |

Device information: 

## Description

**Expected behavior:** 

**Current behavior:** 

```
Launching lib/main.dart on iPhone 15 Pro Max in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                           101.6s
Failed to build iOS app
Error (Xcode): type argument 'nw_proxy_config_t' (aka 'struct nw_proxy_config *') is neither an Objective-C object nor a block type
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:119:46

Parse Issue (Xcode): Could not build module 'WebKit'
/Users/ying/work/app/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:893:8

Could not build the application for the simulator.
Error launching application on iPhone 15 Pro Max.
```

## Steps to reproduce

1. This
2. Than that
3. Then

**Comment by tom-quiltt** [comment]
Might be related to https://github.com/pichillilorenzo/flutter_inappwebview/issues/1735 but I can't do the workaround: https://github.com/pichillilorenzo/flutter_inappwebview/issues/1735#issuecomment-1683434110

**Comment by 9rganizedchaos** [comment]
Completely same issue with me!
The workaround above also doesn't work for me... (even though I tried with `sudo vim ~`)
Can't someone help us...🥲

**Comment by redfish31** [comment]
Hi,

I have the same problem but on a real device (iPhone 12 Pro with iOS 17).

**Comment by redfish31** [comment]
> Might be related to #1735 but I can't do the workaround: [#1735 (comment)](https://github.com/pichillilorenzo/flutter_inappwebview/issues/1735#issuecomment-1683434110)

It's not the same error and it's not fixed the problem for us 😢

**Comment by RMatushkin** [comment]
> > Might be related to #1735 but I can't do the workaround: [#1735 (comment)](https://github.com/pichillilorenzo/flutter_inappwebview/issues/1735#issuecomment-1683434110)
> 
> It's not the same error and it's not fixed the problem for us 😢

Bro, you are most likely doing something wrong. Right now I did what is written in this comment https://github.com/pichillilorenzo/flutter_inappwebview/issues/1735#issuecomment-1699023475, tested it on iOS 17, everything works fine

**Comment by RMatushkin** [comment]
I wrote a small script to make this temporary fix. Please note that the script contains version 15.0 of Xcode. This script works fine in the [Codemagic](https://codemagic.io/) Pre-build script phase.

```shell
#!/bin/bash

echo "Temporary fix for 'WKWebsiteDataStore.h'"

file_path="/Applications/Xcode-15.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/System/Cryptexes/OS/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h"

if [ ! -e "$file_path" ]; then
  echo "File not found: $file_path"
  exit 1
fi

chmod u+w "$file_path"

if [ $? -eq 0 ]; then
  echo "File permissions changed successfully"
else
  echo "Error: Changing file permissions failed"
  exit 1
fi

sed -i '' 's/__IPHONE_OS_VERSION_MAX_ALLOWED >= 170000/__IPHONE_OS_VERSION_MAX_ALLOWED >= 180000/g' "$file_path"

if [ $? -eq 0 ]; then
  echo "Replacement completed successfully"
else
  echo "Error: Replacement encountered an error"
  exit 1
fi

chmod u-w "$file_path"
```

**Comment by eziochiu** [comment]
we add this line `s.ios.deployment_target = '8.0'` solved the problem

**Comment by alarkirikal** [comment]
> I wrote a small script to make this temporary fix. This script works fine in the [Codemagic](https://codemagic.io/) **Pre-build script** phase.
> 
> ```shell
> #!/bin/bash
> 
> echo "Temporary fix for 'WKWebsiteDataStore.h'"
> 
> file_path="/Applications/Xcode-15.0.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/System/Cryptexes/OS/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h"
> 
> if [ ! -e "$file_path" ]; then
>   echo "File not found: $file_path"
>   exit 1
> fi
> 
> chmod u+w "$file_path"
> 
> if [ $? -eq 0 ]; then
>   echo "File permissions changed successfully"
> else
>   echo "Error: Changing file permissions failed"
>   exit 1
> fi
> 
> sed -i '' 's/__IPHONE_OS_VERSION_MAX_ALLOWED >= 170000/__IPHONE_OS_VERSION_MAX_ALLOWED >= 180000/g' "$file_path"
> 
> if [ $? -eq 0 ]; then
>   echo "Replacement completed successfully"
> else
>   echo "Error: Replacement encountered an error"
>   exit 1
> fi
> 
> chmod u-w "$file_path"
> ```

These fixes are great, but this isn't a clean solution, when you have to modify a dependency code on your own in a pre-build step, not to mention about playing around with file ownership..

**Comment by DeanFs** [comment]
I have the same problem

**Comment by redfish31** [comment]
@RMatushkin You're solution is good for the build but it's not working for local debug. When I'm trying to "flutter run ios", I have this error : 

`Error (Xcode): type argument 'nw_proxy_config_t' (aka 'struct nw_proxy_config *') is neither an Objective-C object nor a block type
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:119:46

Parse Issue (Xcode): Could not build module 'WebKit'
/Users/xavier/IdeaProjects/app_mobile/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:893:8

Parse Issue (Xcode): Could not build module 'WebKit'
/Users/xavier/IdeaProjects/app_mobile/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:285:8

Could not build the application for the simulator.
Error launching application on iPad Pro (12.9-inch) (6th generation).`

**Comment by RMatushkin** [comment]
> @RMatushkin You're solution is good for the build but it's not working for local debug. When I'm trying to "flutter run ios", I have this error :
> 
> > Error (Xcode): type argument 'nw_proxy_config_t' (aka 'struct nw_proxy_config *') is neither an Objective-C object nor a block type
> > /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.0.sdk/System/Library/Frameworks/WebKit.framework/Headers/WKWebsiteDataStore.h:119:46
> > Parse Issue (Xcode): Could not build module 'WebKit'
> > /Users/xavier/IdeaProjects/app_mobile/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:893:8
> > Parse Issue (Xcode): Could not build module 'WebKit'
> > /Users/xavier/IdeaProjects/app_mobile/build/ios/Debug-iphonesimulator/flutter_inappwebview/flutter_inappwebview.framework/Headers/flutter_inappwebview-Swift.h:285:8
> > Could not build the application for the simulator.
> > Error launching application on iPad Pro (12.9-inch) (6th generation).

Don't see a problem with it, you can create .sh script locally, and execute it with 'sudo' to have permission for editing. But if you use iOS simulator, you have to edit the same file but in **iPhoneSimulator.platform**.

**Comment by redfish31** [comment]
I'm so stupid ! You're right ! It's working now 🤩

**Comment by GeceGibi** [comment]
Any changes ?

**Comment by nesquikm** [comment]
> Any changes ?

Fixed in #1790

**Comment by roshansrashtasoft** [comment]
![Screenshot 2023-09-26 at 12 24 55](https://github.com/pichillilorenzo/flutter_inappwebview/assets/97586645/85e6e558-055b-4f3c-936f-c609bd771957)

Any Solution I am Stuck 😢 ?

**Comment by kjxbyz** [comment]
```
flutter_inappwebview:
     git:
      url: https://github.com/nesquikm/flutter_inappwebview
      ref: heads/v5.7.2+3_fix_xcode_17
```

**Comment by alarkirikal** [comment]
Or just start using https://pub.dev/packages/webview_flutter instead of referencing some branches on a dependency that hasn't been updated in 9 months (not counting pre-release versions) :)

**Comment by roshansrashtasoft** [comment]
> ```
> flutter_inappwebview:
>      git:
>       url: https://github.com/nesquikm/flutter_inappwebview
>       ref: heads/v5.7.2+3_fix_xcode_17
> ```

I am Already Using the v5.7.2+3 Version.
Any Other Solution?

**Comment by roshansrashtasoft** [comment]
> > ```
> > flutter_inappwebview:
> >      git:
> >       url: https://github.com/nesquikm/flutter_inappwebview
> >       ref: heads/v5.7.2+3_fix_xcode_17
> > ```
> 
> @kjxbyz  I am Already Using the v5.7.2+3 Version. Any Other Solution?

**Comment by kjxbyz** [comment]
@roshansrashtasoft 

```
dependencies:
  flutter:
    sdk: flutter
  flutter_inappwebview:
    git:
      url: https://github.com/nesquikm/flutter_inappwebview
      ref: heads/v5.7.2+3_fix_xcode_17
  flutter_localizations:
    sdk: flutter
  intl: ^0.18.1
```

**Comment by pacharapold** [comment]
> @roshansrashtasoft `pubspec.yaml`
> 
> ```
> dependencies:
>   flutter:
>     sdk: flutter
>   flutter_inappwebview:
>     git:
>       url: https://github.com/nesquikm/flutter_inappwebview
>       ref: heads/v5.7.2+3_fix_xcode_17
>   flutter_localizations:
>     sdk: flutter
>   intl: ^0.18.1
> ```

I attempted to add that solution to the `pubspec.yaml` file, but I encountered the same error. 
Are there any alternative solutions available?

**Comment by kjxbyz** [comment]
Please provide a minimal reproducible example.

**Comment by jiqimao3528** [comment]
> @roshansrashtasoft `pubspec.yaml`
> 
> ```
> dependencies:
>   flutter:
>     sdk: flutter
>   flutter_inappwebview:
>     git:
>       url: https://github.com/nesquikm/flutter_inappwebview
>       ref: heads/v5.7.2+3_fix_xcode_17
>   flutter_localizations:
>     sdk: flutter
>   intl: ^0.18.1
> ```

Thank you, I use this solution, solved the problem

**Comment by felipecastrosales** [comment]
any expectations about the new release @pichillilorenzo?

**Comment by pichillilorenzo** [comment]
Released new version `5.8.0` and `6.0.0-beta.25` with the fix!

**Comment by pengkhieng** [comment]
I have the same problem.
But now it work well just use this command.

flutter pub upgrade
